docs/architecture/adr-028-public-key-addresses.md

```diff
@@ -208,20 +208,29 @@ If we want to create an address for a module account depending on more than one
 btcAtomAMM := address.Module("amm", btc.Addrress() + atom.Address()})
 ```
 
-We can continue the derivation process and can create an address for a submodule account.
+#### Derived Addresses
 
-```
-func Submodule(address []byte, derivationKey []byte) {
-    return Hash("module", address + derivationKey)
+Module address is a special case of more general _derived_ address defined by the `Derive` function:
+
+```go
+func Derive(address []byte, derivationKey []byte) {
+    return Hash(addres, derivationKey)
 }
 ```
 
-NOTE: if `address` is not a hash based address (with `LEN` length) then we should use `LengthPrefix`. An alternative would be to use one `Module` function, which takes a slice of keys and mapped with `LengthPrefix`. For final version we need to validate what's the most common use.
-
 
 **Example**  For a cosmwasm smart-contract address we could use the following construction:
 ```
-smartContractAddr := Submodule(Module("cosmwasm", smartContractsNamespace), smartContractKey)
+smartContractAddr := Derived(Module("cosmwasm", smartContractsNamespace), []{smartContractKey})
+```
+
+We can also define a function which will derive an address based on multiple keys (path). The function is similar to the `NewComposed`, however it doesn't sort the derivation keys:
+
+```go
+func DeriveMulti(address []byte, derivationKeys [][]byte) {
```

Comment:
we could maybe add a changelog entry to this file too for DeriveMulti

Reply:
up

Reply:
will do once we will decide about the changelog updates.

Reply:
I removed this function